src/NServiceBus.Persistence.DynamoDB.Tests/ClientFactory.cs

```diff
@@ -0,0 +1,34 @@
+namespace NServiceBus.Persistence.DynamoDB.Tests
+{
+    using System;
+    using Amazon.DynamoDBv2;
+    using Amazon.Runtime;
+
+    public static class ClientFactory
+    {
+        public static IAmazonDynamoDB CreateDynamoDBClient(Action<AmazonDynamoDBConfig> configure = default)
+        {
+            var credentials = new EnvironmentVariablesAWSCredentials();
+            var config = Create(configure);
+            return new AmazonDynamoDBClient(credentials, config);
+        }
+
+        // Can be removed once https://github.com/aws/aws-sdk-net/issues/1929 is addressed by the team
+        // setting the cache size to 1 will significantly improve the throughput on non-windows OSS while
+        // windows had already 1 as the default.
+        // There might be other occurrences of setting this setting explicitly in the code base. Make sure to remove them
+        // consistently once the issue is addressed.
+        // The method is deliberately generic because this config should be set on any SDK client.
+        static TConfig Create<TConfig>(Action<TConfig> configure = default)
```

Comment:
any reason this method is using generics when we're always using it with `AmazonDynamoDBConfig` anyway?

Reply:
> The method is deliberately generic because this config should be set on any SDK client.

Reply:
I was debating whether this should be generic or not but felt that having it there is a good reminder that we need to set this setting on any client we use. For example if we start adding S3 integration tests (S3 Links are supported on the context type) the client creation would need to use this config too. I do agree though right now it is YAGNI